runtime/compiler/trj9/control/rossa.cpp

```diff
@@ -1136,32 +1136,41 @@ onLoadInternal(
          jitConfig->codeCacheTotalKB = 64 * 1024;
          jitConfig->dataCacheTotalKB = 192 * 1024;
 #endif
+
 #if defined(J9ZTPF)
-#define ZTPF_PERCENTAGE_OF_MEMORY 20;
          /*
           * The z/TPF OS does not have the ability to reserve memory. It allocates whatever it
           * is asked to allocate. Allocate the code cache based on a percentage (20%) of memory
-          * the process is allowed to have; cap the maximum at 256m and use a minimum of 1m for
-          * a small MAXXMMES. It is found in the z/TPF cinfc table entry CMMMMES.
+          * the process is allowed to have; cap the maximum using the default 256m and use a 1m floor.
           * MAXXMMES is specified as the number of 1MB frames.
-          * Create a 2048 KB data cache. This set of default values for the z/TPF os can be
+          * Use the default 2048 KB data cache. The default values for z/TPF can be
           * overriden via the command line interface.
           */
-         uint16_t physMemory = *(uint16_t*)(cinfc_fast(CINFC_CMMMMES) + 8);
-         if (physMemory < 64)
+
+#define ZTPF_CODE_CACHE_DIVISOR 5
+
+         /* Retrieve maximum number of 1MB frames that a z/TPF process can have. */
+         static uint16_t physMemory = *(uint16_t*)((unsigned long int)cinfc_fast(CINFC_CMMMMES) + 8);
+         /* Get 20% of maximum number of 1MB frames to determine default code cache size. */
+         physMemory /= ZTPF_CODE_CACHE_DIVISOR;
```

Comment:
I'll start a different thread here related to the code. @jjohnst-us in my previous review I made a suggestion to make `ZTPF_CODE_CACHE_DIVISOR` a `const` value, not `physMemory`. `physMemory` cannot be const as it is modified in line 1155, but never again. We can make this code more robust, more accurate, and type safe with the following semantically equivalent change:

```
const uint16_t ZTPF_CODE_CACHE_DIVISOR = 5;

/* Retrieve a percentage of the maximum number of 1MB frames that a z/TPF process can have. */
const uint16_t physMemory = *(static_cast<uint16_t*>(cinfc_fast(CINFC_CMMMMES)) + 4) / ZTPF_CODE_CACHE_DIVISOR;
```